cpp/src/arrow/compute/api_scalar.h

```diff
@@ -178,6 +179,20 @@ class ARROW_EXPORT StrptimeOptions : public FunctionOptions {
   TimeUnit::type unit;
 };
 
+class ARROW_EXPORT StrftimeOptions : public FunctionOptions {
+ public:
+  explicit StrftimeOptions(std::string format, std::string timezone);
+  StrftimeOptions();
+  constexpr static char const kTypeName[] = "StrftimeOptions";
+
+  /// The desired format string.
+  std::string format;
+  /// Timezone to output the time in.
+  std::string timezone;
```

Comment:
How does this keyword behave if the type already has a timezone (then a simple conversion I assume) or if the type has no time zone?

(alternative could be to leave out this option and require users first to explicitly convert from one timezone to another (which is a cheap metadata only change) or localize timezone-naive data)

Reply:
Although for local timestamp without timezone, requiring the user to first localize it to a timezone-aware type might give additional overhead, as `strftime` would then convert it back to a local timestamp?

Reply:
I'm not sure we can print timezone naive timestamps without "understanding" them as UTC first anyway (in the `date.h` logic) so I don't think we can get rid of the overhead. But we can choose to do it with one kernel or two.